templates/base.html

```diff
@@ -62,6 +62,19 @@
 var $$ = function(selector) {
   return document.querySelectorAll(selector);
 };
+
+// Note: this version requires Promises, which are in all the
+// latest browsers except IE 11. Edge will ship them.
+// Note: native Promise.defer() is deprecated. Have to create our own.
+function deferred() {
```

Comment:
`defer()` is an anti-pattern:
- http://www.datchley.name/promise-patterns-anti-patterns/#3deferredantipattern
- https://github.com/petkaantonov/bluebird/wiki/Promise-anti-patterns#the-deferred-anti-pattern

I know `defer()`/`deferred()` snuck into the IOWA code base, but it probably shouldn't have. I think you can safely ditch it following the examples in those links.